### PR TITLE
my secret plan to piss off shadow succeeded

### LINKED ISSUE #1043: One of holoporosites is OP
I don't know which one of them and I couldn't find damage data except for https://tgstation13.org/wiki/Holoparasite , so I suppose it must be standard one.

My health before attack: 100%
Defended by HoS standard gear: -10% brute from security jumpsuit for arms/legs (?)
It made 2 hits, first one dismembering my left arm, second one dismembering left leg, resulting in KO I never could get up from - I think next attack has put me in crit.
All this time its owner could do anything else not even participating in combat with me.

IMO it's unbalanced. most of the time traitor can fight against one enemy at time, slipping/stun-proof holoporosite able to deal 1vs1 and even if something goes wrong his ability to pull stunned host is overwhelming.

### PR DESCRIPTION
**Pull request by ghost**
# shadow you dumb motherfucker

The assassin holoparasite now takes double damage from brute and burn, and is not immune to stamina damage.

The assassin holoparasite does less damage out of stealth.

The assassin holoparasite takes slightly less time to re-stealth.

The assassin holoparasite cannot recall for 8 instead of 4 seconds.

Assassin holoparasite does less damage in stealth, especially towards the upper body (to discourage instant decap lel)

Assassin holoparasites DO NOT NEED TO BE MANIFESTED TO ENTER STEALTH

Hopefully fixes #1043
#### Changelog

:cl:
rscadd: Assassin holoparasites are less ridiculous.
rscadd: Assassin holoparasites do not need to be manifested to enter stealth.
/:cl:

### CODE REVIEW

**Comment by Time-Green** [comment]
What is it with you and double damage?
It's ALOT. I know you're not a math king but this is getting ridiculous

**Comment by ghost** [comment]
> not a math king

are you stupid time-green what part of my job do you think does not involve math

@Time-Green jesus fucking christ

okay here's the reason for double damage

they can stealth before manifesting, show up, and fucking cut off your legs and then your head. They're god tier in 1v1.

But if they get caught outside-AKA NOT IN A 1v1-THEY ARE FUCKED.

is that not clear enough to you? ASSASSIN holoparasites are intended to be actively discouraged from fighting in groups/against armed opponents that they don't have the drop on, so getting them fucking killed is a good way to go about it

**Comment by Time-Green** [comment]
I was jokingly refering to the other PR.

Anyway, you convinced me.

**Comment by ShadowDeath6** [comment]
fuck off, adam.

Assassin still sucks massive ass and making them take 10 damage from a punch and killing their host from 5 disabler beams is retarded, because newsflash, all holo transferred damage is converted to brute.
This is a horrible change. 

Also, if you had actually looked at the code for simple mob dismemberment, you would know that NO simple animal can EVER call dismember on head or chest.

**Comment by ghost** [comment]
You sure about that, shadow? I've had xenos cut off my head or slash out my organs before. Assassin is op, you're the only one who doesn't think so so i'm gonna shit all over it

**Comment by ShadowDeath6** [comment]
if(M.melee_damage_type == BRUTE && M.dismember_chance > 0 && dam_zone != "chest")

that's the check, and simplemobs will always avoid head by default due to target picking

**Comment by ShadowDeath6** [comment]
also assassin holos are only OP in the sense that if RNG is in their favor and they get a leg hit while unstealthed, so they can just easyspam you while down, and probably dismembering most of your limbs, and was addressed in https://github.com/yogstation13/yogstation/pull/1060.

**Comment by ghost** [comment]
oh shit you ruined my xenos with your simplemob dismemberment changes at one point

God dammit shadow they used to be able to tear out organs

Let simplemob xenos cut off organs again

Also simplemob xenos have cut off my head in the past

**Comment by ShadowDeath6** [comment]
that was never changed actually, incoming from /tg/ did it differently but that was after our rebase, so I did it differently.

File: code/modules/mob/living/simple_animal/guardian/types/assassin.dm [hunk 1]
```diff
@@ -59,15 +59,12 @@
...
 		updatestealthalert()
 		toggle = FALSE
 	else if(stealthcooldown <= world.time)
-		if(src.loc == summoner)
-			src << "<span class='danger'><B>You have to be manifested to enter stealth!</span></B>"
-			return
-		melee_damage_lower = 50
-		melee_damage_upper = 50
+		melee_damage_lower = 45
+		melee_damage_upper = 35
```
Comment (ShadowDeath6) [inline code/modules/mob/living/simple_animal/guardian/types/assassin.dm]: boi

**Comment by ShadowDeath6** [comment]
:-1: to this PR's idea. It turns a situationally great holo into a borderline useless bullet whore that's less useful being there than not. It also reinforces the lel meta of chaos, standard, or healer holo above all else.

**Comment by Super3222** [comment]
I'd like to hear more from what the players have to say. Recently, it's been exposed that assassin holoparsites are unfairly deadly. I don't feel like we should ignore this. @ShadowDeath6 someone has even told you what happens within crowds and it seems like you have ignored it. Here's their message: 

> I don't know mechanics, but I was dismembered and it didn't feel like "stealth". There was 4 people around, holo comes up to me, hits to death in seconds and continue fighting the rest of people around.

It makes me wonder if they are even generally weak within the public. What are you expecting them to do? Of course damage is instantly transferred to the host, but so what? If it's still able to do 50 brute damage, 100% armor pen, and 50% chance to dismember in about four seconds and escape at will, than I don't think that matters at all.

For example, the bomb holo has a total of 20 seconds of cool down. Once their trap is executed, they have no other choice than to escape otherwise they'll be killed. That is the balancing factor. They're fucked when causing choas in an open area.

Assassin holoparasites don't need to escape. They just need to _wait_. This  dulls the idea of a quick execution, and just creates a war-prepared holo. Correct me if I'm wrong, but that's not the idea.

I find the solution to be increasing their stealth mode cooldown time after an attack. Definitely beyond 4 seconds. My suggestion is a little complex. It's more of a threshold idea, where the next cooldown increases on the amount of attacks they're made recently. After an expired amount of time, it all resets.

So the first attack, 4 second cooldown, a timer that runs for 3 minutes starts. Second attack, fifteen second cooldown. Third attack, it goes up by five. fifth attack, you get the idea.

If the timer runs out, than the cooldown resets to 4 seconds like the first attack. If the cooldown time reaches up to 30 seconds, than it resets automatically. Which means they'd have to make 5 attacks with 90 seconds of CD all together (this is all the cooldowns added up).

Maybe correspondingly, increasing their recall cooldown time as well?

Holoparasites (and possibly their owners) will get a message informing them that they're stealth recharge has returned to normal.

With this system, assassins will still be able to knock out a target. However, in crowds they'll be forced to retreat.

I don't find it soothing to see you gloat about how balanced assassin parasites are right now, because all you've done with them was buffing and buffing and buffing. I don't even want to get into your most recent buff. There's been no real alternative to fighting this type of holoparsite unless "XDXD CLICK FASTER" isn't a meme, and is actually suppose to be taken seriously.

Neither am I siding with this change, because it seems to be rushed and a lot of the changes irrational. I don't think you touched bases with what's actually wrong with holoparsites. Making them die faster isn't the problem. The problem is that if you changed their name to "Commando Type" no one would be able to tell the difference.

Like I said in the beginning, I think we should really look into how it's being used in situations, and the results of it in the game. I'm not surprised that people have come forward to complain, but neither has that made me completely accept that assassin can't be countered. Anyone can come forward and complain about anything. After reviewing a lot of the code and drawing out my own conclusion, I believe it's in an overpowered state right now, and we need to tone it down.

**Comment by Super3222** [comment]
I'd like to hear more from what the players have to say. Recently, it's been exposed that assassin holoparsites are unfairly deadly. I don't feel like we should ignore this. @ShadowDeath6 someone has even told you what happens within crowds and it seems like you have ignored it. Here's their message: 

> I don't know mechanics, but I was dismembered and it didn't feel like "stealth". There was 4 people around, holo comes up to me, hits to death in seconds and continue fighting the rest of people around.

It makes me wonder if they are even generally weak within the public. What are you expecting them to do? Of course damage is instantly transferred to the host, but so what? If it's still able to do 50 brute damage, 100% armor pen, and 50% chance to dismember in about four seconds and escape at will, than I don't think that matters at all.

For example, the bomb holo has a total of 20 seconds of cool down. Once their trap is executed, they have no other choice than to escape otherwise they'll be killed. That is the balancing factor. They're fucked when causing choas in an open area.

Assassin holoparasites don't need to escape. They just need to _wait_ for a few seconds. This  dulls the idea of a quick execution, and just creates a war-prepared holo. Correct me if I'm wrong, but that's not the holo's main idea.

I find the solution to be increasing their stealth mode cooldown time after an attack. Definitely beyond 4 seconds. My suggestion is a little complex. It's more of a threshold idea, where the next cooldown increases on the amount of attacks they're made recently. After an expired amount of time, it all resets.

So the first attack, 4 second cooldown, a timer that runs for 3 minutes starts. Second attack, fifteen second cooldown. Third attack, it goes up by five. fifth attack, you get the idea.

If the timer runs out, than the cooldown resets to 4 seconds like the first attack. If the cooldown time reaches up to 30 seconds, than it resets automatically. Which means they'd have to make 5 attacks with 90 seconds of CD all together (this is all the cooldowns added up).

Maybe correspondingly, increasing their recall cooldown time as well?

Holoparasites (and possibly their owners) will get a message informing them that they're stealth recharge has returned to normal.

With this system, assassins will still be able to knock out a target. However, in crowds they'll be forced to retreat.

I don't find it soothing to see you gloat about how balanced assassin parasites are right now, because all you've done with them was buffing and buffing and buffing. I don't even want to get into your most recent buff. There's been no real alternative to fighting this type of holoparsite unless "XDXD CLICK FASTER" isn't a meme, and is actually suppose to be taken seriously.

Neither am I siding with this change, because it seems to be rushed and a lot of the changes irrational. I don't think you touched bases with what's actually wrong with holoparsites. Making them die faster isn't the problem. The problem is that if you changed their name to "Commando Type" no one would be able to tell the difference.

Like I said in the beginning, I think we should really look into how it's being used in situations, and the results of it in the game. I'm not surprised that people have come forward to complain, but neither has that made me completely accept that assassin can't be countered. Anyone can come forward and complain about anything. After reviewing a lot of the code and drawing out my own conclusion, I believe it's in an overpowered state right now, and we need to tone it down.

**Comment by ShadowDeath6** [comment]
I have a better fix in mind for holos, this seems too severe a nerf to actually be merged.

File: code/modules/mob/living/simple_animal/guardian/types/assassin.dm (hunk 1 above)
Comment (X-TheDark) [reply code/modules/mob/living/simple_animal/guardian/types/assassin.dm]: Funny trivia, despite me fixing it for clarity sakes, this DOES actually work properly.
Doesn't matter if you do `rand(35, 45)` or `rand(45, 35)`, they both work exactly the same.

**Comment by Paragonid** [comment]
Holos are being used as ultimate weapons in most of the situations. As well as shield for the host because they can drag stunned people (that's the best that security typically can do). IMO Dragging should be disabled for holos.
Gun, bombs, chems - you have limited ammo supply for it.
Esword - you have to fight in melee range
Holos - basically unlimited, you can shoot from distance without damage to your holo with stun and let him finish targets in melee. Basically, you don't have to fight at all most of the time, just get some burn/brute patches in rare cases someone hits your holo and watch it kills everyone.
80% of the kills in the game made during stun, due to low projectile speed / click to fire response and high moving speed. Holos never slow down, they don't get stunned, don't get slipped. Those factors make them the most powerful weapons as well as free companions you don't need to work to get (compared to useless mindslave which I love but you have to risk to get one and it's a typical human after all you have to arm etc)

#1071

**Comment by ghost** [comment]
my secret plan to piss off shadow succeeded

**Comment by ShadowDeath6** [comment]
holos did need a nerf in that respect and I was focused on buffing the ones that were worse than the others, hopefully my PR will make some real balance difference

**Pull request closed without merging**